### PR TITLE
#65 service level retry budget

### PR DESCRIPTION
**Pull request by Chris2W**
- Added bucket_retry decorator to utilize TokenBudgetLimiter to limit the amount of retries
- Needing additional proper docs

### CODE REVIEW

File: hyx/retry/manager.py [hunk 1]
```diff
@@ -7,24 +7,37 @@
 from hyx.retry.exceptions import AttemptsExceeded
 from hyx.retry.listeners import RetryListener
 from hyx.retry.typing import AttemptsT, BackoffsT
+from hyx.ratelimit.managers import TokenBucketLimiter
 
+class GlobalRetryLimiter:
```
Comment (roma-glushko) [inline hyx/retry/manager.py:12]: `GlobalRetryLimiter` doesn't seem to do much. Do you think its worth introducing it vs. using `TokenBucketLimiter` (or maybe some abstract class/protocol derived from its signature)?

**Review by roma-glushko — APPROVED** [review]
Looks great to me 👍 
Thank you for contribution Chris!

> Needing additional proper docs

We can add docs in follow up PRs

File: hyx/retry/manager.py (hunk 1 above)
Comment (Chris2W) [reply hyx/retry/manager.py:12]: That's true. GlobalRetryLimiter *could* be extended to take in different limiting algorithms, unless TokenBucketLimiter algorithm is sufficient for retries?

Comment (roma-glushko) [reply hyx/retry/manager.py:12]: Yeah, seems like the signature of the current limiters are good enough for what we demand right now on the retry side. So I would start small by leveraging the existing stuff and then see if we have any case for introducing more sophisticated interfaces

Comment (Chris2W) [reply hyx/retry/manager.py:12]: will do

**Pull request merged**